src/App.js

```diff
@@ -1,15 +1,22 @@
 import { Routes, Route } from 'react-router-dom'
 import Layout from './components/Layout'
-import Login from './components/Login'
 import Dashboard from './components/Dashboard'
 import Profile from './components/Profile'
 import Logout from './components/Logout'
+import Login from './components/Login'
+import useAuth from './useAuth'
 
 function App() {
+	const user = useAuth().auth;
+
+	if(!user){
+		return <Login />
+	}
```

Comment:
This flow works, but it's not optimal and has edge cases. When I log in - the route does not change (from /login to /dashboard for example). When I log out, the route also stays /logout.

Ideally, we need to have [Protecter Routes](https://blog.logrocket.com/complete-guide-authentication-with-react-router-v6/#creating-protected-routes). Then - if someone tries to access /dashboard route and it's not logged in - redirect to /login route.